build/ConsolidateCalamariPackages/SashimiPackageReference.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Serilog;
+
+namespace Calamari.Build.ConsolidateCalamariPackages
+{
+    class SashimiPackageReference : IPackageReference
+    {
+        private readonly Hasher hasher;
+        private readonly BuildPackageReference packageReference;
+
+        public SashimiPackageReference(Hasher hasher, BuildPackageReference packageReference)
+        {
+            this.hasher = hasher;
+            this.packageReference = packageReference;
+        }
+
+        public string Name => packageReference.Name;
+        public string Version => packageReference.Version;
+        public string PackagePath => packageReference.PackagePath;
+
+        public IReadOnlyList<SourceFile> GetSourceFiles(ILogger log)
+        {
+            var extractPath = $"{Path.GetDirectoryName(PackagePath)}/extract/{Name}";
```

Comment:
I've decided to unzip the packages instead of working directly off a zipped package;
This seemed cleaner, and was less "disruptive" to the original logic. Happy to change this back to working with a zipped package if we don't want to unzip.